sympy/matrices/common.py

```diff
@@ -816,37 +834,76 @@ def jordan_block(kls, *args, **kwargs):
         [0, x, 1, 0],
         [0, 0, x, 1],
         [0, 0, 0, x]])
+
+        Creating an alternative Jordan block matrix where `1` is on
+        lower off-diagonal:
+
         >>> Matrix.jordan_block(4, x, band='lower')
         Matrix([
         [x, 0, 0, 0],
         [1, x, 0, 0],
         [0, 1, x, 0],
         [0, 0, 1, x]])
+
+        Creating a Jordan block with keyword arguments
+
         >>> Matrix.jordan_block(size=4, eigenvalue=x)
         Matrix([
         [x, 1, 0, 0],
         [0, x, 1, 0],
         [0, 0, x, 1],
         [0, 0, 0, x]])
-        """
 
-        klass = kwargs.get('cls', kls)
-        size, eigenvalue = None, None
-        if len(args) == 2:
-            size, eigenvalue = args
-        elif len(args) == 1:
-            size = args[0]
-        elif len(args) != 0:
-            raise ValueError("'jordan_block' accepts 0, 1, or 2 arguments, not {}".format(len(args)))
-        rows, cols = kwargs.get('rows', None), kwargs.get('cols', None)
-        size = kwargs.get('size', size)
-        band = kwargs.get('band', 'upper')
-        # allow for a shortened form of `eigenvalue`
-        eigenvalue = kwargs.get('eigenval', eigenvalue)
-        eigenvalue = kwargs.get('eigenvalue', eigenvalue)
+        Creating a rectangular Jordan block:
+
+        >>> Matrix.jordan_block(rows=4, cols=3, eigenvalue=x)
```

Comment:
I would expect a Jordan block to be square.

Reply:
I see Jordan block was added in #12691, and rectangular creation had supported from the point.
But should it really not make any sense?

Reply:
Is this issue resolve?

Reply:
I suppose originally it had allowed only square blocks,
But this 'hidden' feature had been introduced by @siefkenj.

Would rectangular Jordan cells be any useful?
If not, I would want to deprecate this feature.

Reply:
A Jordan block describes a linear mapping of linear (sub)space into itself. It should be square.

Reply:
And also, I wonder how a lower off-digonal Jordan block would be used in factorization.

Reply:
> Would rectangular Jordan cells be any useful?

From what @jksuom is saying, it sounds like there is nobody that would even be using a non-square Jordan block...so I suspect that deprecation would not affect anyone (and could be skipped).

Reply:
I'm not aware of anyone who uses non-square Jordan blocks, however some people prefer upper-triangular and some lower-triangular forms.